src/main.rs

```diff
@@ -43,6 +43,29 @@ impl Display for Commands {
     }
 }
 
+fn handle_tokens(tree: &tree_sitter::Tree, file_data: &str) {
+    // get root node of parsed tree
+    let root_node = tree.root_node();
+    let mut cursor = root_node.walk();
+
+    // iterate over all named nodes in tree
+    println!("Tokens:");
+    let mut stack = vec![(root_node, 0)];
+    while let Some((node, _depth)) = stack.pop() {
```

Comment:
depth is being used so we should drop the leading underscore
```suggestion
    while let Some((node, depth)) = stack.pop() {
```

Reply:
I wonder if first collecting the nodes and then displaying it might be cleaner 🤔 


see `implp From<Node<'_>> for SyntaxNode` here (specifically how the `children` field gets populated) as an example:
https://github.com/mkatychev/topiary/blob/196ef37931fc9edcdf609e2f2c3f94c08d87ae79/topiary-core/src/tree_sitter.rs#L127-L146

The printing is then handled by an `impl Display` on the `SynaxNode`:
https://github.com/mkatychev/topiary/blob/196ef37931fc9edcdf609e2f2c3f94c08d87ae79/topiary-core/src/graphviz.rs#L65-L83

that means that ideally you can do somethin akin to this in `Commands::Tokens`:
```rs
let our_wit_node = OurNode::from(tree.root_node());
println!("{our_wit_node}");
```

Reply:
Yeah, that looks nice!

Reply:
Just expanding on @mkatychev 's comment, it's often much easier and cleaner to define your own types (structures and enums) upfront and design your code around your own types. So when interacting with other libraries (in this case treesitter), you massage the external type into something that is more appropriate for what you need. This at the beginning may "seem" inefficient as you're doing some data conversion but in the context of a larger application often helps you avoid doing unnecessary work multiple times and decouples your code's logic from the underlying library.